### PR TITLE
[WoF] Avoid code duplication and fix a non-translatable string

### LINKED ISSUE #8368: WoF S03: Untranslatable title for the turns over advantage dialog
### Game and System Information

- **Version:** 1.17.25+dev

### Description of the bug

WoF S03 uses a modified copy of the multiplayer turns-over advantage dialog, copied from data/multiplayer/eras.lua to data/campaigns/Winds_of_Fate/lua/lua.lua. It uses strings from textdomain wesnoth-multiplayer, which is probably good, except it tries to add a different title depending on whether the player checks mid-game or when the turns are over. The `dialog^Advantage` string isn't included in any .po file.

Looks like po/CMakeLists.txt and po/SConstruct.txt are good at keeping things to their proper domain, so only Lua files found by po/wesnoth-multiplayer/FINDCFG can put strings in wesnoth-multiplayer. This means only Lua files that are found in data/multiplayer/ or subdirs can add strings to wesnoth-multiplayer.pot.

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CelticMinstrel**
Instead of copy-pasting the multiplayer turns over advantage function, just use that function directly.

This requires making some minor changes to the function for this specific use-case, but I think it's still better than copy-pasting it.

This also fixes #8368.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Apparently I can't request a review from @Jonathan-Kelly, but I can at least ping him.

File: data/multiplayer/eras.lua [hunk 1]
```diff
@@ -97,6 +97,8 @@ res.turns_over_advantage = function()
 	-- if #winning_sides==0, then every side either has no units or has a negative score
 
 	-- po: "Turns Over", meaning "turn limit reached" is the title of the end-of-match summary dialog
-	local a, b = gui.show_popup(_ "dialog^Turns Over", side_comparison)
+	title = title or _ "dialog^Turns Over"
+	local a, b = gui.show_popup(title, side_comparison)
```
Comment (stevecotton) [inline data/multiplayer/eras.lua:101]: Is there any point in having `local a, b` here? It's my fault that they were added, but I can't remember why I added them.

**Comment by gfgtdf** [comment]
Have you tested this? The first few lines of `res.turns_over_advantage` check  explicitly whether the current game is a mp campaign and does nothing  in that case.

**Comment by stevecotton** [comment]
There's two different places in WoF S03 that call this, please test them both. One's the menu item added in `[event]name=prestart` and the other's the one in `[event]name=turns over`.

File: data/multiplayer/eras.lua (hunk 1 above)
Comment (CelticMinstrel) [reply data/multiplayer/eras.lua:101]: I'm really not sure why it's there. `gui.show_popup` doesn't return anything, though, so it seems both variables will always be `nil` anyway.

**Comment by CelticMinstrel** [comment]
> The first few lines of `res.turns_over_advantage` check explicitly whether the current game is a mp campaign and does nothing in that case.

Looks like it skips the check if a certain WML variable is set, so I just set it to true before calling.

> One's the menu item added in `[event]name=prestart` and the other's the one in `[event]name=turns over`.

I don't know how I missed the second one…

**Comment by stevecotton** [comment]
Please do more testing, at the moment the menu item does this:
```
error wml: Failed to load required package: multiplayer/lua/eras.lua
error scripting/lua: [string "..."]:4: attempt to index a nil value (local 'eras')
```

As adding the title option is a slight API change anyway, I think it might be worth going further:
* keep the existing behavior of `eras.turn_advantage_dialog` checking wml.variables["show_turns_over_advantage"], but
* move most of the code to a new `eras.show_turn_advantage_dialog(title)`, which doesn't check that WML variable

**Comment by CelticMinstrel** [comment]
> As adding the title option is a slight API change anyway

Personally, I don't consider that file to be part of the API, since it's not in the data/lua folder nor the data/ai folder. But I did go ahead and split the function apart, like you suggested.

**Comment by Jonathan-Kelly** [comment]
By the way, you can test the functionality quickly by loading scenario 3 from the start, right clicking anywhere on the map and then selecting the `Show Advantage` option from the right click menu.

Currently, doing so throws this error for me:

`error scripting/lua: multiplayer/eras.lua:123: variable '_' must be assigned before being used`

**Comment by stevecotton** [comment]
I tested in multiplayer, set up a two player game with turn limit of 1. After both sides ended their turn:
* Player 2 saw "Side 2 has the advantage"
* Player 2 saw "You have been **defeated**"

**Comment by stevecotton** [comment]
Okay, I retested on 1.16.11+dev and the behavior is the same there, so it's not a regression.

**Review by stevecotton — APPROVED** [review]

**Pull request merged**